### PR TITLE
Check Julia compat separately.

### PR DESCRIPTION
**Pull request by GunnarFarneback**
This separates the compat checks of Julia from compat of other packages.
1. To make bad Julia compat more visible in the automerge report.
2. To make additional checks, specifically that the upper bound of Julia is not 2 or higher, which is unreasonable at this time.

Currently General has 12 packages with an upper bound of "2" or "2.0". All of these are probably misunderstandings of hyphen ranges, thinking that the upper end is exclusive, e.g. "1.5 - 2.0".

Additional tests of this have not been added yet.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
See also: https://github.com/JuliaRegistries/General/pull/35311

File: src/AutoMerge/guidelines.jl
Comment (ericphanson) [inline src/AutoMerge/guidelines.jl:36]:
```suggestion
                    return false, "The compat entry for `julia` is unbounded."
```

File: src/AutoMerge/guidelines.jl
Comment (ericphanson) [inline src/AutoMerge/guidelines.jl:39]:
```suggestion
                    return false, "The compat entry for `julia` has an upper bound of 2 or higher."
```

File: src/AutoMerge/guidelines.jl
Comment (ericphanson) [inline src/AutoMerge/guidelines.jl:44]:
```suggestion
                    return false, "The compat entry for `julia` doesn't include any 1.x version."
```

File: src/AutoMerge/guidelines.jl
Comment (ericphanson) [inline src/AutoMerge/guidelines.jl:52]:
```suggestion
    return false, "There is no compat entry for `julia`."
```

**Review by ericphanson — COMMENTED** [review]
Makes sense to me to separate it out.

I thought the wording of the reject messages here could possibly be confusing; e.g. "Julia has an unbounded compat entry" to me sounds like it's a property of Julia itself, not the package here. So I've suggested changing them to "The compat entry for `julia`..." to try to be more clear. I also went for lowercase `julia` instead of Julia to try to better match the key for the compat entry.

File: src/AutoMerge/guidelines.jl
Comment (GunnarFarneback) [reply src/AutoMerge/guidelines.jl:29]: Failure to remove a copy/pasted comment.

File: src/AutoMerge/guidelines.jl
Comment (GunnarFarneback) [reply src/AutoMerge/guidelines.jl:35]: It's the type used for the elements of a `VersionNumber`, https://github.com/JuliaLang/julia/blob/4e176ac96565a7e72f16689a3d6a5d465e35148a/base/version.jl#L25-L28.

The use here is to create the version range [4294967295.0.0 - *], which we intersect with the compat range to determine whether the latter is unbounded. It's slightly cheating but way easier than going deeper into the internals of `VersionSpec` et al to figure that out.

Comment (ericphanson) [reply src/AutoMerge/guidelines.jl:35]: ok, thanks for the explanation. Does `[4294967295.0.0 - *]` represent any upper-bounded range?

Comment (GunnarFarneback) [reply src/AutoMerge/guidelines.jl:35]: It's on the philosophic end of things but this test does give a false positive for e.g. `"1 - 4294967295"`, which technically is an upper-bounded range.

**Comment by GunnarFarneback** [comment]
Are there currently any tests for missing or unbounded compat?

**Comment by ericphanson** [comment]
I don't think there's any integration tests for that, unfortunately

**Comment by DilumAluthge** [comment]
It would be nice to have tests for that. But those can be added in future pull requests. Doesn't need to hold up this PR.

**Comment by GunnarFarneback** [comment]
Well, let's see what bors thinks about this then.

bors try

**Comment by GunnarFarneback** [comment]
bors try

**Comment by DilumAluthge** [comment]
Even though Bors didn't post a comment, if you look at the commit, everything is green.

Once @ericphanson has approved this, you can merge it.

**Review by ericphanson — APPROVED** [review]

**Comment by GunnarFarneback** [comment]
We want a version bump also I suppose. New minor?

**Comment by DilumAluthge** [comment]
Master is currently at a dev version. I have some breaking changes I want to make soon, so let's just leave master at the current dev version number

**Comment by GunnarFarneback** [comment]
bors merge

**Pull request merged**
